packages/integration-tests/src/test-projects/data-storage-auth-with-triggers-ts/amplify/data/resource.ts

```diff
@@ -38,7 +38,14 @@ const schema = a.schema({
     .arguments({ content: a.string() })
     .returns(a.ref('EchoResponse'))
     .authorization([a.allow.private()])
-    .function('echo'),
+    .handler(
```

Comment:
is the old way still supported? If so, maybe we should add this as another resolver and keep the existing one

Reply:
It's marked as deprecated now and will be removed before GA. 

However, you can still pass a string to the new handler, e.g. 
```ts
getPostDetails: a
    .query()
    .arguments({ postId: a.string() })
    .handler([
      a.handler.function('myFunc'),
    ])
```
In which case you'll have to configure via `DataProps.functions`

Reply:
We can test both here

Reply:
Is there a use case where you would have to specify the name and the separate mapping? Maybe we remove the `functions` prop entirely from `defineData` and just allow passing the references through the schema?

Reply:
Updated

Reply:
I don't think it's relevant to Amplify Gen2 specifically. We kept the string param in place as an escape hatch for the Amplify CDK construct

Reply:
In that case I'd be in favor of removing the `functions` map entirely. Any objections @sobolk?

Reply:
None. less is better.

Reply:
After offline discussion, we decided to wait and make this change in a separate PR